### PR TITLE
Expand switch description and examples to deprecate comments.

### PR DESCRIPTION
**Pull request by Crell**
I cleaned out most of the comments on the `switch` page.  These are the bits worth including.

### CODE REVIEW

File: language/control-structures/switch.xml [hunk 1]
```diff
@@ -181,6 +189,77 @@ switch ($i) {
...
+    case $start - 3:
+        print "C";
+        break;
+    case $start - 4:
+        print "C";
+        break;
+}
+?>
+]]>
+   </programlisting>
```
Comment (cmb69) [inline language/control-structures/switch.xml:232]: I think we should mention the output of the example above. Not necessarily by adding an `&example.outputs;` section, but rather mentioning that it is "B".

File: language/control-structures/switch.xml [hunk 2]
```diff
@@ -181,6 +189,77 @@ switch ($i) {
...
+        print "A";
+        break;
+    case $start - 2:
+        print "B";
+        break;
+    case $start - 3:
+        print "C";
+        break;
+    case $start - 4:
+        print "C";
```
Comment (cmb69) [inline language/control-structures/switch.xml:227]: Just for the sake of clarity:
```suggestion
        print "D";
```

File: language/control-structures/switch.xml [hunk 3]
```diff
@@ -181,6 +189,77 @@ switch ($i) {
...
+        break;
+}
+?>
+]]>
+   </programlisting>
+  </informalexample>
+ </para>
+ <para>
+  For more complex comparisons, use the value <literal>true</literal> as the switch value.
```
Comment (cmb69) [inline language/control-structures/switch.xml:236]: I'd go with passive voice here ("the value … can be used"). Analogous for the next line.

File: language/control-structures/switch.xml [hunk 4]
```diff
@@ -181,6 +189,77 @@ switch ($i) {
...
+        print "A";
+        break;
+    case $start - $offset === 2:
+        print "B";
+        break;
+    case $start - $offset === 3:
+        print "C";
+        break;
+    case $start - $offset === 4:
+        print "C";
```
Comment (cmb69) [inline language/control-structures/switch.xml:256]:
```suggestion
        print "D";
```

File: language/control-structures/switch.xml [hunk 5]
```diff
@@ -181,6 +189,77 @@ switch ($i) {
...
+        break;
+}
+?>
+]]>
+   </programlisting>
```
Comment (cmb69) [inline language/control-structures/switch.xml:261]: Again, what is the output?

File: language/control-structures/switch.xml [hunk 6]
```diff
@@ -66,6 +66,14 @@ switch ($i) {
...
 <![CDATA[
 <?php
+if ($i == 'apple') {
+    echo "i is apple";
+} elseif ($i == 'bar') {
+    echo "i is bar";
+} elseif ($i == 'cake') {
+    echo "i is cake";
+}
+
```
Comment (cmb69) [inline language/control-structures/switch.xml:76]: That should have an explanation; maybe we should revert both parts to show the switch first, then have a comment "is the same as:", and then the if-elseif construct. We should probably do the same with the first example, although that has at least some introduction paragraph explaining what's going on there.

**Review by cmb69 — COMMENTED** [review]
Thank you!

File: language/control-structures/switch.xml (hunk 6 above)
Comment (Crell) [reply language/control-structures/switch.xml:76]: The intro paragraph specifically says it applies to both code samples.  The second one was just missing the equivalent code.  (It's all kinda weird and silly, frankly.)  I'll fiddle with it a bit.

Comment (Crell) [reply language/control-structures/switch.xml]: Not clearly, as I read it as being both examples.:-)  But in the end I concluded that having 2 examples offers exactly zero benefit over 1, at least for such similar examples, so I removed one and reworded it accordingly.

**Review by cmb69 — APPROVED** [review]
Thank you! Looks good to me.

There is the already existing "It's possible to use a semicolon instead of a colon after a case like:". And while it is indeed possible, I tend to discourage its use, since I'm pretty sure that this was only done to support the alternative syntax `<?php case "whatever"?>` where the trailing semicolon could be omitted.

**Comment by Crell** [comment]
Er, not sure what the semicolon business is about?  That's already there in the docs, and while I agree no one should use it, it's worth noting that it exists, as is already done.  That's unrelated to this PR.

**Pull request merged**